### PR TITLE
vip: add querier for weight vote and vip

### PR DESCRIPTION
**Pull request by JSHan94**
- add `get_bridge_infos` for querying all whitelisted bridge infos
- add `get_challenge_by_stage` for challenges in specific stage

## Summary by CodeRabbit

- **New Features**
  - Introduced a function to retrieve bridge information.
  - Added a function to get challenge responses based on a specified stage.

- **Enhancements**
  - Included a new constant for default stage periods.
  - Updated various structures and functions to incorporate stage periods.

### CODE REVIEW

File: precompile/modules/initia_stdlib/sources/vip/weight_vote.move [hunk 1]
```diff
@@ -1177,6 +1177,48 @@ module initia_std::vip_weight_vote {
...
+        let module_store = borrow_global<ModuleStore>(@initia_std);
+        let iter = table::iter(
+            &module_store.challenges,
+            option::none(),
+            option::none(),
+            1
+        );
+
+        let challenge_responses = vector::empty<ChallengeResponse>();
+        loop {
```
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/vip/weight_vote.move:1191]: how about storing challenge key to stage data as `Option` to avoid iteration?

Comment (JSHan94) [reply precompile/modules/initia_stdlib/sources/vip/weight_vote.move:1191]: seems challenge query cost will not be expensive i guess

File: precompile/modules/initia_stdlib/sources/vip/vip.move [hunk 2]
```diff
@@ -70,6 +71,8 @@ module initia_std::vip {
         // global stage
         stage: u64,
         // governance-defined vesting period in stage unit
+        stage_period: u64,
+        // governance-defined vesting period in stage unit
```
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/vip/vip.move:75]: seems redundant

**Review by beer-1 — APPROVED** [review]
LGTM

**Pull request merged**